sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -179,13 +179,17 @@ class FilesElement extends Polymer.Element {
     clearInterval(this._fileListRefreshIntervalHandle);
   }
 
-  _getNotebookUrlPrefix() {
-    let prefix = location.protocol + '//' + location.host + '/';
+  async _getNotebookUrlPrefix() {
+    // Notebooks that are stored on the VM requires the basepath.
+    let basepath = await ApiManager.getBasePath();
```

Comment:
I would've preferred to keep the base path hidden in `ApiManager` and instead expose a `getNotebookUrlPrefix` there, but up to you. :)